internal/cmd/commands/dev/dev.go

```diff
@@ -824,74 +824,93 @@ func (c *Command) Run(args []string) int {
 	c.opsServer = opsServer
 	c.opsServer.Start()
 
-	// Wait for shutdown
-	shutdownTriggered := false
+	var shutdownCompleted atm.Bool
+	shutdownTriggerCount := uint32(0)
 
-	// Add a force-shutdown goroutine to consume more interrupts
-	abortForceShutdownCh := make(chan struct{})
-	defer close(abortForceShutdownCh)
+	var workerShutdownOnce sync.Once
+	workerShutdownFunc := func() {
+		if err := c.worker.Shutdown(); err != nil {
+			c.UI.Error(fmt.Errorf("Error shutting down worker: %w", err).Error())
+		}
+	}
+	workerGracefulShutdownFunc := func() {
+		if !c.flagWorkerAuthStorageSkipCleanup && c.worker.WorkerAuthStorage != nil {
```

Comment:
I would put this storage cleanup in `workerShutdownFunc` after the `Shutdown` call. That way in case something tries to access the files while in the process of shutting down we won't have unexpected behavior.

Reply:
Done, thanks!